src/Exception/DirectoryCreationException.php

```diff
@@ -17,6 +17,6 @@ class DirectoryCreationException extends IOException
      */
     public function __construct($path, \Exception $previous = null)
     {
-        parent::__construct('Failed to create directory', $path, 0, $previous);
+        parent::__construct('Failed to create directory: ' . $path, $path, 0, $previous);
     }
```

Comment:
Just noting for down the track, but I tend to favour [Symfony's coding standard](http://symfony.com/doc/current/contributing/code/standards.html#structure) on exceptions:
  - Exception message strings should be concatenated using sprintf.